h2/src/main/org/h2/index/FunctionCursorResultSet.java

```diff
@@ -30,9 +30,7 @@ boolean nextImpl() {
             int columnCount = result.getVisibleColumnCount();
             Value[] currentRow = result.currentRow();
             values = new Value[columnCount];
-            for (int i = 0; i < columnCount; i++) {
-                values[i] = currentRow[i];
-            }
+            System.arraycopy(currentRow, 0, values, 0, columnCount);
```

Comment:
Arrays.copyOf(T[], int) is a better choice here.

Reply:
Yes, thank you. Fixed